overlord/snapstate/snapstate.go

```diff
@@ -1361,17 +1365,95 @@ var ValidateRefreshes func(st *state.State, refreshes []*snap.Info, ignoreValida
 // UpdateMany updates everything from the given list of names that the
 // store says is updateable. If the list is empty, update everything.
 // Note that the state must be locked by the caller.
-func UpdateMany(ctx context.Context, st *state.State, names []string, userID int, flags *Flags) ([]string, []*state.TaskSet, error) {
-	return updateManyFiltered(ctx, st, names, userID, nil, flags, "")
+func UpdateMany(ctx context.Context, st *state.State, names []string, revOpts []*RevisionOptions, userID int, flags *Flags) ([]string, []*state.TaskSet, error) {
```

Comment:
we should split out the changes to UpdateMany and InstallMany to its own PR. This PR is not that big but things are fairly subtle and those changes should be a bit more obvious, also this has quite a mechanical impact that we would isolate that way.

Reply:
Yes, this is just a draft to validate the approach since it changed from the previous draft. But once we're satisfied I'll split this off. I'll also have to add a spread test coverage